src/hooks/useTx.tsx

```diff
@@ -20,16 +21,30 @@ export const useTx = () => {
         })
       console.log('RegisterOperator')
       if (extension.data) {
-        const result = await api.tx.domains
-          .registerOperator(registration.domainId, registration.amountToStake, {
-            signingKey: registration.signingKey,
-            minimumNominatorStake: registration.minimumNominatorStake,
-            nominationTax: registration.nominatorTax
+        try {
+          const hash = await api.tx.domains
+            .registerOperator(registration.domainId, registration.amountToStake, {
+              signingKey: registration.signingKey,
+              minimumNominatorStake: registration.minimumNominatorStake,
+              nominationTax: registration.nominatorTax
+            })
+            .signAndSend(extension.data?.defaultAccount.address, { signer: injectedExtension.signer })
```

Comment:
I don't believe the return of this is actually an indication of success. It may make sense to wait for it to be included in a block before doing the success toast.

Reply:
Waiting for it to appear in a block takes a long time, so I like the idea of providing some sort of a success message upon submission. 
Perhaps we can tweak the message itself, saying something like "transaction was submitted successfully? You will see the change after the next epoch if the registration was successful"? What do you think?

Reply:
Yes, as mention in engineering channel, I'm not able to run a farmer atm and get any rewards, therefore it's hard for me to actually test making transactions.

But yes, I'm thinking of changing the message and then add the tx hash to a list of tx hash to watch in the state manager, so we can keep watching for the different hash and give a final success toast when they get minted.

Reply:
> Waiting for it to appear in a block takes a long time, so I like the idea of providing some sort of a success message upon submission.
It shouldn't take long to include in a block. Having a message stating it won't be effective until the next epoch will be good too. @marc-aurele-besner see [here](https://polkadot.js.org/docs/api/start/api.tx.subs) for how to subscribe to the transaction events. You can wrap in a promise that can be resolved.

Reply:
![AllTransactionsSuccess](https://github.com/subspace/staking-interface/assets/82244926/9797feb7-6247-4012-bb05-7a1b92812d28)
![PendingTx](https://github.com/subspace/staking-interface/assets/82244926/b3540ede-3c66-467c-89b9-83659e035d6b)

Reply:
I'm not sure it is still capturing the events how I was envisioning but we can come back to it later.